tests/Unit/Type/data/custom-query-type.php

```diff
@@ -0,0 +1,27 @@
+<?php
+declare(strict_types=1);
+
+namespace My\Test\Extension;
+
+use TYPO3\CMS\Extbase\DomainObject\AbstractEntity;
+use TYPO3\CMS\Extbase\Persistence\QueryInterface;
+use function PHPStan\Testing\assertType;
+
+class SomeOtherModel extends AbstractEntity {}
```

Comment:
Also not sure if this will break at some time as this class could again be used in a different test with this namespace. But i think we can solve that by using a more unique namespace for the test itself.

I would also prefer to keep the Extbase default practice and name the model like the repository. So maybe use `MyModel` and `MyModelRepository` instead.

Reply:
But the point of this test is to prove that it works if I _don't_ follow the Extbase convention, so that's not what I want here.
We can rename the repository to MyModelRepository (and add a stub class MyModel as well), but the SomeOtherModel has to stay to keep the test working as intended.

OTOH adding MyModel would maybe also be detrimental to the value of the test, since we want to prove that it also works if we don't stick to the model/repository name convention.

Reply:
Makes sense ;)

Reply:
I renamed the repo and added a comment why the model does not follow the naming convention.